container/go/pkg/compat/overrideConfig.go

```diff
@@ -0,0 +1,545 @@
+	for _, port := range overrideInfo.Ports {
+		match, err := regexp.MatchString("[0-9]+/(tcp|udp)", port)
+		if err != nil {
+			return errors.Wrapf(err, "failed to successfully match regex to %s", port)
+		}
+		if match {
+			// the port spec has the form 80/tcp, 1234/udp so simply use it as the key.
+			overrideInfo.ConfigFile.Config.ExposedPorts[port] = struct{}{}
+		} else {
+			// assume tcp
+			overrideInfo.ConfigFile.Config.ExposedPorts[port+"/tcp"] = struct{}{}
+		}
+	}
+
+	return nil
+}
+
+// updateVolumes modifies the config's volumes based on the input volumes.
+func updateVolumes(overrideInfo *OverrideConfigOpts) error {
+	for _, volume := range overrideInfo.Volumes {
+		overrideInfo.ConfigFile.Config.Volumes[volume] = struct{}{}
+	}
+
+	return nil
+}
+
+// updateConfigLayeres modifies the config's layers metadata based on the input layer IDs and history.
+func updateConfigLayers(overrideInfo *OverrideConfigOpts, layerDigests []string, creationTime time.Time) error {
+	// diffIDs are ordered from bottom-most to top-most.
+	diffIDs := []v1.Hash{}
+	if len(overrideInfo.ConfigFile.RootFS.DiffIDs) > 0 {
+		diffIDs = overrideInfo.ConfigFile.RootFS.DiffIDs
+	}
+	if len(overrideInfo.Layer) > 0 {
+		var diffIDToAdd v1.Hash
+		for _, diffID := range layerDigests {
+			if diffID != emptySHA256Digest() {
+				diffIDToAdd = v1.Hash{Algorithm: "sha256", Hex: diffID}
+				diffIDs = append(diffIDs, diffIDToAdd)
+			}
+		}
+		overrideInfo.ConfigFile.RootFS = v1.RootFS{Type: "layers", DiffIDs: diffIDs}
+
+		// length of history is expected to match the length of diff_ids.
+		history := []v1.History{}
+		if len(overrideInfo.ConfigFile.History) > 0 {
+			history = overrideInfo.ConfigFile.History
+		}
+		var historyToAdd v1.History
+
+		for _, l := range layerDigests {
+			var createdBy = overrideInfo.CreatedBy
+			var authorToAdd = overrideInfo.Author
+
+			if err := overrideInfo.validate(); err != nil {
+				return errors.Wrapf(err, "unable to create a new config because the given options to override the existing image config/generate new config failed validation")
+			}
+
+			historyToAdd = v1.History{
+				Author:    authorToAdd,
+				Created:   v1.Time{creationTime},
+				CreatedBy: createdBy,
+			}
+			if l == emptySHA256Digest() {
+				historyToAdd.EmptyLayer = true
+			}
+			// prepend to history list.
+			history = append([]v1.History{historyToAdd}, history...)
+		}
+		overrideInfo.ConfigFile.History = history
+	}
+
+	return nil
+}
+
+// OverrideImageConfig updates the current image config file to reflect the given changes.
+func OverrideImageConfig(overrideInfo *OverrideConfigOpts) error {
+	overrideInfo.ConfigFile.Author = "Bazel"
+	overrideInfo.ConfigFile.OS = overrideInfo.OperatingSystem
+	overrideInfo.ConfigFile.Architecture = defaultProcArch
+
+	var err error
+
+	// creationTime is the RFC 3339 formatted time derived from createTime input.
+	creationTime, err := time.Parse(time.RFC3339, defaultTimestamp)
```

Comment:
This value is always overridden by the line `if creationTime, err = getCreationTime(overrideInfo); err != nil {` so it can be removed.

You probably need to update the `getCreationTime` function to return `time.Parse(time.RFC3339, defaultTimestamp)` if the override value is blank.